### ISSUE TITLE
rattler-build 0.6.0 breaks cross-python activation scripts

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/rattler-build#410]
```yml
# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json

package:
  name: test-package
  version: '0.1.0'
build:
  script:
    - ls -lah
requirements:
  build:
    - if: build_platform != target_platform
      then:
        - python
        - cross-python_${{ target_platform }}
```

```yml
on:
  push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: mamba-org/setup-micromamba@v1
        with:
          environment-name: test
          create-args: rattler-build=0.6.0
      - run: rattler-build build -r conda.recipe/recipe.yaml --target-platform=linux-aarch64 --no-test
        shell: bash -el {0}
```

I get the following error:

```

Resolving for environment specs:
 - python
 - cross-python_linux-aarch64
...

┌───────────────────────────────────────────┬────────────┬────────────────────┬─────────────┬────────────┐
│ Package                                   ┆ Version    ┆ Build              ┆ Channel     ┆ Size       │
╞═══════════════════════════════════════════╪════════════╪════════════════════╪═════════════╪════════════╡
│ cross-python_linux-aarch64                ┆ 3.12       ┆ 41_cpython         ┆ conda-forge ┆ 12.01 KiB  │
│ python                                    ┆ 3.12.0     ┆ hab00c5b_0_cpython ┆ conda-forge ┆ 30.64 MiB  │
│ xz                                        ┆ 5.2.6      ┆ h166bdaf_0         ┆ conda-forge ┆ 408.56 KiB │
│ readline                                  ┆ 8.2        ┆ h8228510_1         ┆ conda-forge ┆ 274.86 KiB │
│ libuuid                                   ┆ 2.38.1     ┆ h0b41bf4_0         ┆ conda-forge ┆ 32.81 KiB  │
│ libexpat                                  ┆ 2.5.0      ┆ hcb278e6_1         ┆ conda-forge ┆ 76.15 KiB  │
│ libzlib                                   ┆ 1.2.13     ┆ hd590300_5         ┆ conda-forge ┆ 60.14 KiB  │
│ libnsl                                    ┆ 2.0.1      ┆ hd590300_0         ┆ conda-forge ┆ 32.62 KiB  │
│ ncurses                                   ┆ 6.4        ┆ h59595ed_2         ┆ conda-forge ┆ 863.71 KiB │
│ tk                                        ┆ 8.6.13     ┆ noxft_h4845f30_101 ┆ conda-forge ┆ 3.17 MiB   │
│ openssl                                   ┆ 3.2.0      ┆ hd590300_1         ┆ conda-forge ┆ 2.72 MiB   │
│ libsqlite                                 ┆ 3.44.2     ┆ h2797004_0         ┆ conda-forge ┆ 826.01 KiB │
│ bzip2                                     ┆ 1.0.8      ┆ hd590300_5         ┆ conda-forge ┆ 248.27 KiB │
│ ld_impl_linux-64                          ┆ 2.40       ┆ h41732ed_0         ┆ conda-forge ┆ 688.18 KiB │
│ libffi                                    ┆ 3.4.2      ┆ h7f98852_5         ┆ conda-forge ┆ 56.93 KiB  │
│ rsync                                     ┆ 3.2.7      ┆ h70740c4_0         ┆ conda-forge ┆ 329.97 KiB │
│ libiconv                                  ┆ 1.17       ┆ h166bdaf_0         ┆ conda-forge ┆ 1.38 MiB   │
│ lz4-c                                     ┆ 1.9.4      ┆ hcb278e6_0         ┆ conda-forge ┆ 140.04 KiB │
│ xxhash                                    ┆ 0.8.0      ┆ h7f98852_3         ┆ conda-forge ┆ 85.75 KiB  │
│ popt                                      ┆ 1.16       ┆ h0b475e3_2002      ┆ conda-forge ┆ 184.42 KiB │
│ zstd                                      ┆ 1.5.5      ┆ hfc55251_0         ┆ conda-forge ┆ 532.42 KiB │
│ sed                                       ┆ 4.8        ┆ he412f7d_0         ┆ conda-forge ┆ 264.42 KiB │
│ crossenv                                  ┆ 1.4.0      ┆ pyhd8ed1ab_0       ┆ conda-forge ┆ 29.71 KiB  │
│ libgcc-ng                                 ┆ 13.2.0     ┆ h807b86a_3         ┆ conda-forge ┆ 755.50 KiB │
│ _libgcc_mutex                             ┆ 0.1        ┆ conda_forge        ┆ conda-forge ┆ 2.50 KiB   │
│ _openmp_mutex                             ┆ 4.5        ┆ 2_gnu              ┆ conda-forge ┆ 23.07 KiB  │
│ libgomp                                   ┆ 13.2.0     ┆ h807b86a_3         ┆ conda-forge ┆ 411.95 KiB │
│ libstdcxx-ng                              ┆ 13.2.0     ┆ h7e041cc_3         ┆ conda-forge ┆ 3.66 MiB   │
│ tzdata                                    ┆ 2023c      ┆ h71feb2d_0         ┆ conda-forge ┆ 114.82 KiB │
│ ca-certificates                           ┆ 2023.11.17 ┆ hbcca054_0         ┆ conda-forge ┆ 150.50 KiB │
│ gcc_linux-aarch64                         ┆ 13.2.0     ┆ h3cf286d_2         ┆ conda-forge ┆ 29.67 KiB  │
│ binutils_linux-aarch64                    ┆ 2.40       ┆ h19b5abf_2         ┆ conda-forge ┆ 27.45 KiB  │
│ binutils_impl_linux-aarch64               ┆ 2.40       ┆ h40ea8b1_0         ┆ conda-forge ┆ 5.35 MiB   │
│ ld_impl_linux-aarch64                     ┆ 2.40       ┆ hc365e21_0         ┆ conda-forge ┆ 742.26 KiB │
│ gcc_impl_linux-64                         ┆ 13.2.0     ┆ h338b0a0_3         ┆ conda-forge ┆ 50.25 MiB  │
│ libsanitizer                              ┆ 13.2.0     ┆ h7e041cc_3         ┆ conda-forge ┆ 3.88 MiB   │
│ libgcc-devel_linux-64                     ┆ 13.2.0     ┆ ha9c7c90_103       ┆ conda-forge ┆ 2.46 MiB   │
│ binutils_impl_linux-64                    ┆ 2.40       ┆ hf600244_0         ┆ conda-forge ┆ 5.16 MiB   │
│ gcc_impl_linux-aarch64                    ┆ 13.2.0     ┆ h22d4421_3         ┆ conda-forge ┆ 336.23 MiB │
│ libgcc-devel_linux-aarch64                ┆ 13.2.0     ┆ h66a9a87_3         ┆ conda-forge ┆ 1.94 MiB   │
│ sysroot_linux-aarch64                     ┆ 2.17       ┆ h5b4a56d_13        ┆ conda-forge ┆ 15.54 MiB  │
│ kernel-headers_linux-aarch64              ┆ 4.18.0     ┆ h5b4a56d_13        ┆ conda-forge ┆ 1.11 MiB   │
│ _sysroot_linux-aarch64_curr_repodata_hack ┆ 4          ┆ h57d6b7b_13        ┆ conda-forge ┆ 20.31 KiB  │
│ sysroot_linux-64                          ┆ 2.12       ┆ he073ed8_16        ┆ conda-forge ┆ 14.57 MiB  │
│ kernel-headers_linux-64                   ┆ 2.6.32     ┆ he073ed8_16        ┆ conda-forge ┆ 692.39 KiB │
└───────────────────────────────────────────┴────────────┴────────────────────┴─────────────┴────────────┘

✔ Successfully updated the environment

Resolving for environment specs:
 - python 3.12.* *_cpython

┌──────────────────┬────────────┬────────────────────┬─────────────┬────────────┐
│ Package          ┆ Version    ┆ Build              ┆ Channel     ┆ Size       │
╞══════════════════╪════════════╪════════════════════╪═════════════╪════════════╡
│ python           ┆ 3.12.0     ┆ hab00c5b_0_cpython ┆ conda-forge ┆ 30.64 MiB  │
│ xz               ┆ 5.2.6      ┆ h166bdaf_0         ┆ conda-forge ┆ 408.56 KiB │
│ readline         ┆ 8.2        ┆ h8228510_1         ┆ conda-forge ┆ 274.86 KiB │
│ libuuid          ┆ 2.38.1     ┆ h0b41bf4_0         ┆ conda-forge ┆ 32.81 KiB  │
│ libexpat         ┆ 2.5.0      ┆ hcb278e6_1         ┆ conda-forge ┆ 76.15 KiB  │
│ libzlib          ┆ 1.2.13     ┆ hd590300_5         ┆ conda-forge ┆ 60.14 KiB  │
│ libnsl           ┆ 2.0.1      ┆ hd590300_0         ┆ conda-forge ┆ 32.62 KiB  │
│ ncurses          ┆ 6.4        ┆ h59595ed_2         ┆ conda-forge ┆ 863.71 KiB │
│ tk               ┆ 8.6.13     ┆ noxft_h4845f30_101 ┆ conda-forge ┆ 3.17 MiB   │
│ openssl          ┆ 3.2.0      ┆ hd590300_1         ┆ conda-forge ┆ 2.72 MiB   │
│ libsqlite        ┆ 3.44.2     ┆ h2797004_0         ┆ conda-forge ┆ 826.01 KiB │
│ bzip2            ┆ 1.0.8      ┆ hd590300_5         ┆ conda-forge ┆ 248.27 KiB │
│ ld_impl_linux-64 ┆ 2.40       ┆ h41732ed_0         ┆ conda-forge ┆ 688.18 KiB │
│ libffi           ┆ 3.4.2      ┆ h7f98852_5         ┆ conda-forge ┆ 56.93 KiB  │
│ libgcc-ng        ┆ 13.2.0     ┆ h807b86a_3         ┆ conda-forge ┆ 755.50 KiB │
│ _libgcc_mutex    ┆ 0.1        ┆ conda_forge        ┆ conda-forge ┆ 2.50 KiB   │
│ _openmp_mutex    ┆ 4.5        ┆ 2_gnu              ┆ conda-forge ┆ 23.07 KiB  │
│ libgomp          ┆ 13.2.0     ┆ h807b86a_3         ┆ conda-forge ┆ 411.95 KiB │
│ tzdata           ┆ 2023c      ┆ h71feb2d_0         ┆ conda-forge ┆ 114.82 KiB │
│ ca-certificates  ┆ 2023.11.17 ┆ hbcca054_0         ┆ conda-forge ┆ 150.50 KiB │
└──────────────────┴────────────┴────────────────────┴─────────────┴────────────┘
✔ Successfully updated the environment
Work dir: "/home/runner/work/ci-test/ci-test/output/bld/rattler-build_test-package_1702135006/work"
Build script: "/home/runner/work/ci-test/ci-test/output/bld/rattler-build_test-package_1702134671/work/conda_build.sh"
INFO: activate-binutils_linux-aarch64.sh made the following environmental changes:
...
+_CONDA_PYTHON_SYSCONFIGDATA_NAME=_sysconfigdata_aarch64_conda_cos7_linux_gnu
+build_alias=x86_64-conda-linux-gnu
+host_alias=aarch64-conda-linux-gnu
-BUILD=x86_64-conda_cos6-linux-gnu
-CFLAGS=
-LDFLAGS=
-QEMU_LD_PREFIX=
Setting up cross-python
ERROR: [Errno 2] No such file or directory: '/home/runner/work/ci-test/ci-test/output/bld/rattler-build_test-package_1702134671/host_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.12/_sysconfigdata_aarch64_conda_cos7_linux_gnu.py'
Error:   × Build failed

Error: Process completed with exit code 1.
```

When I replace `rattler-build=0.6.0` with `rattler-build=0.5.2`, everything works.

**Comment by wolfv** [comment]
I think it could be related to setting c and cxxflags with the latest version. I'll confirm and remove that again

**Comment by wolfv** [comment]
I debugged a bit and I can reproduce. I think the "real" issue is that we are now executing the scripts with `-x`/`-e`. However, I am not sure if we should allow this to fail or wether this is an actual problem with something else.

I need to double check where this file is supposed to come from.

**Comment by wolfv** [comment]
With the older, rattler-build, do you also see something like:


```
Setting up cross-python
ERROR: [Errno 2] No such file or directory: '/Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/host_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.11/_sysconfigdata_x86_64_apple_darwin13_4_0.py'
cp: /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/host_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.11/_sysconfigdata_x86_64_apple_darwin13_4_0.py: No such file or directory
cp: /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/cross/bin/python: No such file or directory
sed: can't read /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/bin/cross-python: No such file or directory
sed: can't read /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/bin/cross-python: No such file or directory
sed: can't read /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/bin/cross-python: No such file or directory
rm: /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/build/bin/python: No such file or directory
ln: /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/build/bin/python: No such file or directory
ln: /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/lib/python3.11/site-packages: No such file or directory
ln: /Users/wolfv/Programs/rattler-build/output/bld/rattler-build_test-package_1702374059/build_env/venv/lib/sitecustomize.py: No such file or directory
Finished setting up cross-python
```

**Comment by wolfv** [comment]
This is fixed with #411
